src/Artifacts.jl

```diff
@@ -311,19 +312,33 @@ function download_artifact(
 
     # Did we get what we expected?  If not, freak out.
     if calc_hash.bytes != tree_hash.bytes
-        msg  = "Tree Hash Mismatch!\n"
-        msg *= "  Expected git-tree-sha1:   $(bytes2hex(tree_hash.bytes))\n"
-        msg *= "  Calculated git-tree-sha1: $(bytes2hex(calc_hash.bytes))"
-        # Since tree hash calculation is still broken on some systems, e.g. Pkg.jl#1860,
-        # and Pkg.jl#2317, we allow setting JULIA_PKG_IGNORE_HASHES=1 to ignore the
-        # error and move the artifact to the expected location and return true
-        ignore_hash = Base.get_bool_env("JULIA_PKG_IGNORE_HASHES", false)
+        msg = """
+        Tree Hash Mismatch!
+          Expected git-tree-sha1:   $(bytes2hex(tree_hash.bytes))
+          Calculated git-tree-sha1: $(bytes2hex(calc_hash.bytes))
+        """
+        # actual and expected artifiact paths
+        src = artifact_path(calc_hash; honor_overrides=false)
+        dst = artifact_path(tree_hash; honor_overrides=false)
+        # Since tree hash calculation is rather fragile and file system dependent,
+        # we allow setting JULIA_PKG_IGNORE_HASHES=1 to ignore the error and move
+        # the artifact to the expected location and return true
+        if get(ENV, "JULIA_PKG_IGNORE_HASHES", "") == ""
+            # default: false except Windows users who can't symlink
+            ignore_hash = Sys.iswindows() &&
+                !mktempdir(can_symlink, dirname(src))
+        else
+            ignore_hash = Base.get_bool_env("JULIA_PKG_IGNORE_HASHES", false)
+            ignore_hash === nothing && @error(
+                "Invalid ENV[\"JULIA_PKG_IGNORE_HASHES\"] value",
+                ENV["JULIA_PKG_IGNORE_HASHES"],
+            )
+            ignore_hash = something(ignore_hash, false)
+        end
         if ignore_hash
             msg *= "\n\$JULIA_PKG_IGNORE_HASHES is set to 1: ignoring error and moving artifact to the expected location"
             @error(msg)
```

Comment:
```suggestion
        msg = """
        Tree Hash Mismatch!
          Expected git-tree-sha1:   $(bytes2hex(tree_hash.bytes))
          Calculated git-tree-sha1: $(bytes2hex(calc_hash.bytes))
        """
        # actual and expected artifiact paths
        src = artifact_path(calc_hash; honor_overrides=false)
        dst = artifact_path(tree_hash; honor_overrides=false)
        # Since tree hash calculation is rather fragile and file system dependent,
        # we allow setting JULIA_PKG_IGNORE_HASHES=1 to ignore the error and move
        # the artifact to the expected location and return true
        if get(ENV, "JULIA_PKG_IGNORE_HASHES", "") == ""
            # default: false except Windows users who can't symlink
            ignore_hash = Sys.iswindows() &&
                !mktempdir(can_symlink, dirname(src))
            if ignore_hash
                msg *= "\nWindows cannot make symlinks which can cause hash mismatches. Enable Developer Mode to allow symlinks. Ignoring hash mismatch."
            end
        else
            ignore_hash = Base.get_bool_env("JULIA_PKG_IGNORE_HASHES", false)
            ignore_hash === nothing && @error(
                "Invalid ENV[\"JULIA_PKG_IGNORE_HASHES\"] value",
                ENV["JULIA_PKG_IGNORE_HASHES"],
            )
            ignore_hash = something(ignore_hash, false)
            if ignore_hash 
                msg *= "\n\$JULIA_PKG_IGNORE_HASHES is set to $(repr(ENV["JULIA_PKG_IGNORE_HASHES"])): ignoring error and moving artifact to the expected location"
            end
        end
        if ignore_hash
            
            @error(msg)
```

Reply:
Otherwise the messaging makes it seem that the ENV var has been set when it hasn't been